pkg/controller/controller.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	framework "github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+)
+
+// Controller represents a long-lived process that watches objects in the system and can respond to events.  Plugins
+// implement factory methods for creating framework.Controller instances, such as those created by framework.NewInformer()
+// or framework.New(config).
+type Controller interface {
+	// Controllers returns many framework.Controller objects completely configured and ready to run
+	// This function is called once by the framework but should still be idempotent.
+	Controllers() ([]*framework.Controller, error)
```

Comment:
@lavalamp I admit this naming is clunky, but I think correct (though we can find a better name).

The PluginMgr will run all the controllers, keep their stop channels, be able to report status, allow graceful shutdown, etc.  I don't think the controllers themselves need Run.  The plugins need to return N controllers, as you say, because an entire controller may contain many control loops.  

This ```Controllers()``` func is more like volume.NewBuilder or NewCleaner.   It is a factory meant to return instances from the plugin, which the plugin would then run.  PluginMgr reduces the duplication of every controller having their own Run method.